src/pdsparsemat.jl

```diff
@@ -57,31 +58,31 @@ LinearAlgebra.sqrt(A::PDSparseMat) = PDMat(sqrt(Hermitian(Matrix(A))))
 
 ### whiten and unwhiten
 
-function whiten!(r::StridedVecOrMat, a::PDSparseMat, x::StridedVecOrMat)
+function whiten!(r::AbstractVecOrMat, a::PDSparseMat, x::AbstractVecOrMat)
     r[:] = sparse(chol_lower(a.chol)) \ x
     return r
 end
 
-function unwhiten!(r::StridedVecOrMat, a::PDSparseMat, x::StridedVecOrMat)
+function unwhiten!(r::AbstractVecOrMat, a::PDSparseMat, x::AbstractVecOrMat)
     r[:] = sparse(chol_lower(a.chol)) * x
     return r
 end
 
 
 ### quadratic forms
 
-quad(a::PDSparseMat, x::StridedVector) = dot(x, a * x)
-invquad(a::PDSparseMat, x::StridedVector) = dot(x, a \ x)
+quad(a::PDSparseMat, x::AbstractVector) = dot(x, a * x)
+invquad(a::PDSparseMat, x::AbstractVector) = dot(x, a \ x)
 
-function quad!(r::AbstractArray, a::PDSparseMat, x::StridedMatrix)
-    for i in 1:size(x, 2)
+function quad!(r::AbstractArray, a::PDSparseMat, x::AbstractMatrix)
+    for i in axes(x, 2)
```

Comment:
Also here i might not be a valid index for r.

Reply:
fixed check